proteus/mprans/SpatialTools.py

```diff
@@ -1390,7 +1385,7 @@ def assembleAuxiliaryVariables(domain):
     -----
     Should be called after assembleGeometry
     """
-    domain.auxiliaryVariables = []
+    domain.auxiliaryVariables = {'twp': [], 'vof': []}
```

Comment:
I think I like where this is going [1], but I'm not entirely sure I understand how these auxiliaryVariables are being set up.

We start with two collections of auxiliaryVariables, 'twp' and 'vof', which I assume are for the twp_navier_stokes and vof models respectively.  Should we have ones for the other models used as well?  On a brief glance, I at least have a model which brings in auxiliaryVariables from `redist_n.py`, and I wouldn't be surprised if other models have the same problem.

(Maybe it would be nice to have a `getAuxiliaryVariables(*key name*)` method for the domain, which can return `[ ]` if the dictionary does not contain an element [2] )

But anyway, for each shape we keep setting `aux = domain.auxiliaryVariables` (which starts as the empty-ish dictionary we initialized above), and then adding `aux['twp']` values and... then we have `aux[-1]` which we seem to be making some assumptions about (line 1400, if I read these numbers right, sets `body = aux[-1]` and then immediately starts making broad assumptions about body's components).  I'm just not sure what happened to 'vof' (or if it's just there for later expansion), and how the -1 is meant to work after resetting domain.auxiliaryVariables above.

(After looking at it more, I've figured out that the -1 part is just referring back to what is now the 'twp' - I'm guessing it just hasn't been changed yet [but can be without breaking something down the line]?  That eases my confusion a bit, although I still have some other questions I've kept in this)

[1] A dictionary is just what I was looking to implement for gauges.
[2] This would allow each `*_n.py` file to say `auxiliaryVariables = domain.getAuxiliaryVariables(" * ")` and not worry about it breaking if there's nothing for it.  Then the user can throw in gauges (or other `*_n.py` specific auxiliary variables) and toggle them on and off from the main .py file without worrying about changing the specific `*_n.py` files each time - while also being able to affect each and every possible `*_n.py` file they might have, not just `twp` and `vof` (and we wouldn't have to add on new empty dictionary elements for every possible `*_n.py` file.